### PR TITLE
Propagate local properties to broadcast execs

### LINKED ISSUE #9464: [AUD] Audit [SPARK-44897][SQL] Propagating local properties to subquery broadcastexec
**Is your feature request related to a problem? Please describe.**

Spark added in https://github.com/apache/spark/commit/2f4a71252cc

I am not 100% sure exactly which properties are not being propagated, but we do have essentially the same code that was touched in our GpuBroadcastExchangeExec

https://github.com/NVIDIA/spark-rapids/blob/30e5a8f537ee0e40ba713807d1d53df7b267109e/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala#L378

### PR DESCRIPTION
**Pull request by thirtiseven**
Closes #9464 

This PR adds propagating local properties to `GpuBroadcastExchangeExecBase` and `GpuSubqueryBroadcastExec`.

https://github.com/apache/spark/commit/2f4a71252cc Added propagating local properties to subquery broadcast exec, because the local properties of subquery broadcast exec (which are usually propagated by broadcast exchange exec) to the broadcast threads would be wrong in some cases. And we have the same code in the plugin.

The test case in the commit message will fail because we didn't propagate local properties to `GpuBroadcastExchangeExecBase` so I added it. But after I adding it, the test case will pass no matter if the propagating local properties to `GpuSubqueryBroadcastExec` is added and no matter if I added the `Thread.sleep` as the guide from the Spark commit.

I personnally think the spark commit is reasonable and also applicable for plugin so I added it in this PR, but I haven't found a good way to test it. 

As the spark commit message said, It is difficult to write a unit test that reproduces this behavior because usually `BroadcastExchangeExec` is the first computing the broadcast variable.

BTW, `GpuBroadcastToRowExec` already implements propagation, we may want to keep it and `GpuSubqueryBroadcastExec` consistent if propagation is really unnecessary.
https://github.com/NVIDIA/spark-rapids/blob/525c73e4396ae71e82aa8cf4f3da48733a907fce/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastToRowExec.scala#L65

This PR also fixes a nit from my previous commit.

### CODE REVIEW

File: integration_tests/src/main/python/generate_expr_test.py [hunk 1]
```diff
@@ -226,8 +226,11 @@ def test_stack():
 # gpu stack not guarantee to produce the same output order as Spark does
 @ignore_order(local=True)
 def test_stack_mixed_types():
+    base_gens = [byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen, string_gen, 
+                  boolean_gen, date_gen, timestamp_gen, null_gen, DecimalGen(precision=7, scale=3),
+                  DecimalGen(precision=12, scale=2), DecimalGen(precision=20, scale=2)]
     data_gen = StructGen([['child'+str(ind), sub_gen] for ind, sub_gen in 
-                          enumerate(all_basic_gens + decimal_gens)], nullable=False)
+                          enumerate(base_gens)], nullable=False)
```
Comment (revans2) [inline integration_tests/src/main/python/generate_expr_test.py:233]: Is there a reason this test changed? I get that we wanted to make this change for stack anyways. I'm just not sure combining it with an unrelated PR is a good thing.

**Review by revans2 — DISMISSED** [review]

**Comment by revans2** [comment]
Oops it looks like 312 and 313 do not like the code change. We might have to put in some shims to make this work properly.

File: integration_tests/src/main/python/generate_expr_test.py (hunk 1 above)
Comment (thirtiseven) [reply integration_tests/src/main/python/generate_expr_test.py:233]: Ok got it, I will avoid combining unrelated things in one PR in the future. Thanks for the reminding.

**Review by revans2 — APPROVED** [review]

**Comment by revans2** [comment]
build

**Pull request merged**